### PR TITLE
KendoUI Context menu component & behavior

### PR DESCRIPTION
**Pull request by martin-g**
A simple extensions of KendoUI Menu(Behavior).

### CODE REVIEW

**Comment by sebfz1** [comment]
Thanks you very much Martin, it looks perfect!

For the sample purpose, I prefer the feedback panel is declared on each page, because only the leaf page's code is displayed... I will change this with the merge/review/reformat/etc, please don't bother with that... ;)

Thanks again & best regards,
Sebastien

File: wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/widget/menu/MenuBehavior.java [hunk 1]
```diff
@@ -53,7 +53,17 @@ public MenuBehavior(String selector)
 		super(selector, METHOD);
 	}
 
-	/**
+    /**
+     * Constructor
+     *
+     * @param selector the html selector (ie: "#myId")
+     */
+    protected MenuBehavior(String selector, String method)
```
Comment (sebfz1) [inline wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/widget/menu/MenuBehavior.java]: As it is for internal purpose only, wouldn't be better to have a package visibility only?

File: wicket-jquery-ui-samples/src/main/java/com/googlecode/wicket/jquery/ui/samples/pages/kendo/menu/KendoContextMenuPage.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html xmlns:wicket="http://wicket.apache.org">
+<head>
+<wicket:head>
+	<title>Wicket - jQuery UI: kendo-ui context menu</title>
+</wicket:head>
+</head>
+<body>
+<wicket:extend>
+	<div wicket:id="menu"></div>
```
Comment (sebfz1) [inline wicket-jquery-ui-samples/src/main/java/com/googlecode/wicket/jquery/ui/samples/pages/kendo/menu/KendoContextMenuPage.html]: - [ ] @sebfz1 - Add text like: "right-click anywhere to display the context menu..."

**Comment by sebfz1** [comment]
*Houston...*

In the sample page `target.add(this)` does not have any effect: the state of the items are not reflected (change title, desactivate).
The root cause is: on 'kendoContextMenu()' the "#root" is moved away from the "#menu" to a newly created ".k-animation-container" div.
`target.add(this)` actually (re)add the menu and the underlying root, but it does not replace the previous/moved root... 

Maybe tomorrow with a fresh brain...

screenshot from 2015-05-08 03 17 53

**Comment by sebfz1** [comment]
Maybe a candidate for #destroy... will check tomorrow...
http://docs.telerik.com/kendo-ui/api/javascript/ui/contextmenu#methods-destroy

File: wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/widget/menu/MenuBehavior.java (hunk 1 above)
Comment (martin-g) [inline wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/widget/menu/MenuBehavior.java]: Agreed!

File: wicket-jquery-ui-samples/src/main/java/com/googlecode/wicket/jquery/ui/samples/pages/kendo/menu/KendoContextMenuPage.html (hunk 2 above)
Comment (martin-g) [inline wicket-jquery-ui-samples/src/main/java/com/googlecode/wicket/jquery/ui/samples/pages/kendo/menu/KendoContextMenuPage.html]: Done!

**Comment by martin-g** [comment]
I've added `"target": "#menu-target"` setting. Now the context menu opens only when the text is right clicked. I think this solved the problem with the HTML leak. Please confirm!

**Comment by martin-g** [comment]
Moved the feedback panel back to the sample pages!
I should have done this in a branch so it is easier to merge to 7.x... :-/
Please check whether everything is OK now and I'll cherry-pick the commits.

**Comment by sebfz1** [comment]
Hi Martin,

Many thanks for the updates! Everything look ok! :)

It seems that #destroy does the trick, even I realize afterward that "This method does not remove the ContextMenu element from DOM.". So there is a hint I didn't catch yet...

I think #destroy can benefit to other widget, even if there is no problem so far. I will try to move it up in the hierarchy in a near future...

Regarding #onOpen & #isOpenEventEnabled, there are usually defined in an interface (ie: IContextMenuListener) shared by both the behavior and the component. I can handle it no worry.

Also for the alignment with wicket7.x, I can handle it. I will have a few think to update (formatting, some minor renaming/refactoring) to keep full consistency with the rest of the API.

Please let me know when you are done with the context menu, so I enter on stage :)

**Comment by martin-g** [comment]
> Also for the alignment with wicket7.x, I can handle it. I will have a few think to update (formatting, some minor renaming/refactoring) to keep full consistency with the rest of the API.

Please do this in a separate commit so I can use it to fix my code style settings.

**Comment by martin-g** [comment]
> Please let me know when you are done with the context menu, so I enter on stage :)

The context menu seems to work fine for me with the latest improvements.
I'm playing now with stacked Window-s here. I'll let you know if I need improvements in this area.

**Comment by sebfz1** [comment]
> Please do this in a separate commit so I can use it to fix my code style settings.

Do you mean a commit just about the formatting?

> The context menu seems to work fine for me with the latest improvements.

Agree!

> I'm playing now with stacked Window-s here

Have fun :)

**Comment by martin-g** [comment]
> Do you mean a commit just about the formatting?

Yes. So I can see easily the exact changes I have to do (I don't use Eclipse and I don't want to guess what its settings in the XML files actually mean :-) ).

**Pull request merged**